### PR TITLE
Split: Setting up Ubuntu for Programming

### PR DESCRIPTION
**Pull request by theflametrooper**
Still todo:
- [x] Check filenames are correct.
- [x] Add intros to each section.
- [x] Add links to previous, related articles in each section.

### CODE REVIEW

File: Ubuntu-Customising.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Customising Ubuntu
+
```
Comment (abhisekp) [inline Ubuntu-Customising.md]: Add an intro text and then heading 2

File: Ubuntu-Downloading-Text-Editors-and-IDEs.MD [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Downloading Text Editors and IDEs
+
```
Comment (abhisekp) [inline Ubuntu-Downloading-Text-Editors-and-IDEs.MD]: Add an intro here and add heading 2

**Comment by abhisekp** [comment]
All the splitted articles must have intro text upto first heading 2

**Comment by theflametrooper** [comment]
@abhisekp or @Rafase282 How will we add links back to the main `Ubuntu` article or `previous`/`next` article.

**Comment by Rafase282** [comment]
@theflametrooper use [this](https://github.com/FreeCodeCamp/wiki/issues/816#issuecomment-216679394).

**Comment by Rafase282** [comment]
@theflametrooper besides the `##` please ass a title first.

File: Ubuntu-Installing-DevTools-and-Modern-Web-Browsers.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Installing DevTools and modern web browsers
+
+We show you how to install various programming langauges, devtools and web browsers on Ubuntu.
+
+##
```
Comment (Rafase282) [inline Ubuntu-Installing-DevTools-and-Modern-Web-Browsers.md]: `## Widely used tools`

File: Ubuntu-Downloading-Text-Editors-and-IDEs.MD [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+We discuss popular Text Editors and IDEs and how to install them on Ubuntu.
+
+##
```
Comment (Rafase282) [inline Ubuntu-Downloading-Text-Editors-and-IDEs.MD]: `## Popular Choices`

File: Ubuntu-Alternatives-to-popular-Windows-and-Mac-Software.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Alternatives to popular Windows and Mac software
```
Comment (Rafase282) [inline Ubuntu-Alternatives-to-popular-Windows-and-Mac-Software.md]: Ubuntu-Software-Alternatives.md for the filename.

**Comment by Rafase282** [comment]
Made some minor suggestions, other than that, looks good to me.

File: Ubuntu-Installing-DevTools-and-Modern-Web-Browsers.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
+# Installing DevTools and modern web browsers
```
Comment (Rafase282) [inline Ubuntu-Installing-DevTools-and-Modern-Web-Browsers.md]: Just `Ubuntu-Installing-DevTools.md` should do. The version of firefox and chromium are devtools and modem browsers at the same time.

File: Ubuntu-Downloading-Text-Editors-and-IDEs.MD [hunk 7]
```diff
@@ -0,0 +1,74 @@
+# Downloading Text Editors and IDEs
```
Comment (Rafase282) [inline Ubuntu-Downloading-Text-Editors-and-IDEs.MD]: `Ubuntu-Downloading-Text-Editors-and-IDEs.md` lowercase `.md`

**Pull request merged**